source/write-operations/bulk-writes.txt

```diff
@@ -1,58 +1,347 @@
+
+If any of the write operations fail, the {+driver-short+} raises a
+``BulkWriteError`` and does not perform any further operations.
+``BulkWriteError`` provides a ``details`` attribute that includes the operation
+that failed, and details about the exception.
+
+.. note::
+
+   When the {+driver-short+} runs a bulk operation, it uses the ``writeConcern`` of the
+   collection in which the operation is running. The driver reports all write
+   concern errors after attempting all operations, regardless of execution order. 
+
+Customize Bulk Write Operations
+-------------------------------
+
+The ``BulkWriteOptions`` class contains methods that modify
+the behavior of the ``bulkWrite()`` method. To use the ``BulkWriteOptions``
+class, construct a new instance of the class, then call one or more of its methods
+to modify the delete operation. You can chain these method calls together.
+To modify the behavior of the delete operation, pass the class instance and
+chained method calls as the last argument to the
+``bulkWrite()`` method.
+
+You can use the following methods in the ``BulkWriteOptions`` class
+to modify a delete method. All methods are optional.
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Property
+     - Description
+
+   * - ``bypassDocumentValidation(Boolean bypassDocumentValidation)``
+     - | Specifies whether the bulk write operation bypasses document validation. This lets you 
+         perform write operations on documents that don't meet the schema validation requirements, if any 
+         exist. For more information about schema validation, see :manual:`Schema
+         Validation </core/schema-validation/#schema-validation>` in the MongoDB
+         Server manual.
+
+   * - ``comment(Bson comment)``
+     - | Attaches a ``Bson`` comment to the operation. For more information, see the :manual:`insert command
+         fields </reference/command/insert/#command-fields>` guide in the
+         MongoDB Server manual.  
+
+   * - ``comment(String comment)``
+     - | Attaches a ``String`` comment to the operation. For more information, see the :manual:`insert command
+         fields </reference/command/insert/#command-fields>` guide in the
+         MongoDB Server manual.  
+
+   * - ``let(Bson variables)``
+     - | Specifies a map of parameter names and values. Values must be constant or closed
+         expressions that don't reference document fields. For more information,
+         see the :manual:`let statement
+         </reference/command/update/#std-label-update-let-syntax>` in the
+         MongoDB Server manual.
+
+   * - ``ordered(Boolean ordered)``
+     - | If set to ``True``, the driver sends documents to the
+         server in the order provided. If an error occurs, the driver
+         and server cancel all remaining write operations.
+       | Defaults to ``True``. 
+
+The following example calls the ``bulkWrite()`` method from the preceding example, with the ``ordered`` option set
```

Comment:
S: to avoid "with":
```suggestion
The following example calls the ``bulkWrite()`` method from the preceding example but sets the ``ordered`` option
```